kong/config_service.go

```diff
@@ -37,29 +38,25 @@ func (c *ConfigService) ReloadDeclarativeRawConfig(
 	}
 	req, err := c.client.NewRequest("POST", "/config", sendConfigParams{CheckHash: checkHashI}, config)
 	if err != nil {
-		return fmt.Errorf("creating new HTTP request for /config: %w", err)
+		return []byte{}, fmt.Errorf("creating new HTTP request for /config: %w", err)
 	}
 
 	resp, err := c.client.DoRAW(ctx, req)
 	if err != nil {
-		return fmt.Errorf("failed posting new config to /config: %w", err)
+		return []byte{}, fmt.Errorf("failed posting new config to /config: %w", err)
 	}
 	defer resp.Body.Close()
 
+	var b []byte
 	if resp.StatusCode < 200 || resp.StatusCode >= 400 {
-		b, err := io.ReadAll(resp.Body)
+		b, err = io.ReadAll(resp.Body)
 		if err != nil {
-			return fmt.Errorf(
-				"failed posting new config to /config: got status code %d (and failed to read the response body): %w",
-				resp.StatusCode, err,
-			)
+			return []byte{},
```

Comment:
When I look at this now I believe returning a `nil` makes no sense. Why allocate an empty slice?

Actually: after looking at https://groups.google.com/g/golang-nuts/c/bqeemNtZkb4 it seems that there's little to no difference. I'd prefer to return `nil` to make it explicit though. WDYT?

Reply:
No reason for the empty slice. I just don't naturally think in terms of acceptable Go conversions, so I see `[]byte` return signature, I return `[]byte{}` if I have nothing, since that's more clearly a `[]byte`.

I changed it and then read the thread and realized that it doesn't matter. But hey, it doesn't matter.